### PR TITLE
Prepare for new Play Store listing

### PR DESCRIPTION
**Pull request by msfjarvis**
## :loudspeaker: Type of change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [x] Refactoring

## :scroll: Description
Cleanup Gradle configuration, setup caching in GitHub Actions, and update package name to `dev.msfjarvis.aps`.

## :bulb: Motivation and Context
Discussed in #597

## :green_heart: How did you test it?
https://github.com/android-password-store/Android-Password-Store/commit/30adddd412afed203f327b2ad92ad7a6e79349fb/checks?check_suite_id=363564694

## :pencil: Checklist

- [x] I ran `./gradlew spotlessApply` before submitting the PR
- [x] I reviewed submitted code

## :crystal_ball: Next steps
Prepare Play Store listing

### CODE REVIEW

**Comment by msfjarvis** [comment]
WIP in general and also blocking on #600 for now

**Comment by mimi89999** [comment]
Could you please keep the same packagename for F-Droid releases?

**Comment by msfjarvis** [comment]
> Could you please keep the same `applicationId` for F-Droid releases?

F-Droid's not a priority for me and I don't understand their weird metadata. If someone wants to retain F-Droid builds then they can submit a merge request upstream when this PR is merged with whatever changes are necessary.

File: dependencies.gradle [hunk 1]
```diff
@@ -3,12 +3,13 @@
  * SPDX-License-Identifier: GPL-3.0-only
  */
 ext.versions = [
-    minSdk: 21,
-    targetSdk: 29,
-    compileSdk: 29,
+    minSdk: 23,
```
Comment (Skrilltrax) [inline dependencies.gradle]: minSdk 23 :thinking:

Comment (msfjarvis) [reply dependencies.gradle]: I'm moving all my projects to minSdk 23, API 21/22 have broken drawable behaviour, missing crypto and other problems that are not worth our time. It'll also allow us to fix #526 for all users in a consistent fashion.

Comment (Skrilltrax) [reply dependencies.gradle]: Oh okay :+1:

**Pull request merged**